Assets/SceneChangeDoor.cs

```diff
@@ -0,0 +1,59 @@
+using System;
+using UnityEngine;
+using UnityEngine.InputSystem;
+using UnityEngine.SceneManagement;
+
+public class SceneChangeDoor : MonoBehaviour
+{
+    public GameObject player;
+    public string targetScene;
+    private bool playerInInteractionArea = false;
+    private InputControls input = null;
+
+    private void Awake()
+    {
+        input = new InputControls();
+        input.Enable();
+    }
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        
+    }
+
+    private void OnTriggerEnter(Collider other)
```

Comment:
I believe that OnTriggerEnter might not trigger if an object spawns inside the trigger area of the other object. E.g. if the player spawns inside a trigger area.

If that's the case, it might be better to use OnTriggerStay https://docs.unity3d.com/ScriptReference/Collider.OnTriggerStay.html